tests/Internal/HttpListener/HttpListener.Tests.ps1

```diff
@@ -0,0 +1,192 @@
+Describe 'HttpListener Tests' {
+    
+    BeforeAll {
+        Import-Module -Name "$PSScriptRoot\HttpListener.psm1" -Verbose 
+    }
+
+    AfterAll {
+        Remove-Module -Name HttpListener -Verbose
+    }
+  
+    It 'Should start the listener at the selected port' {
+        # Verify the function is imported
+        Get-Command Start-HttpListener | Should -Not -BeNullOrEmpty
+        
+        $port = 9000
+        $url = "http://127.0.0.1:$port/okta-powerShell/?test=response"
+
+        try {
+
+            { Start-HTTPListener -Port $port } | Should -Not -Throw
+            
+            # Send a test request to verify the server is running
+            Start-Sleep -Seconds 2 # Allow server to start
+
+            $response = Invoke-WebRequest -Uri $url
+            $response.StatusCode | Should -Be 200
+        }
+        finally {
+            Stop-HTTPListener -Port $port
+        }
+   }
+
+   It 'Should stop the listener' {    
+        $port = 9000
+        $url = "http://127.0.0.1:$port/okta-powerShell/?test=response"
+
+        { Start-HTTPListener -Port $port } | Should -Not -Throw
+            
+        # Send a test request to verify the server is running
+        Start-Sleep -Seconds 2 # Allow server to start
+
+        $response = Invoke-WebRequest -Uri $url 
+        $response.StatusCode | Should -Be 200
+
+        Stop-HTTPListener -Port $port
+
+       { Invoke-WebRequest -Uri $url -SkipHttpErrorCheck } | Should -Throw
+    }
+
+   It 'Should return tailored responses' {
+        $port = 9000
+        $headers = @{'X-Okta-Request-Id'='foo';'X-Rate-Limit-Reset'='bar'} | ConvertTo-Json -Compress 
+        $url = "http://127.0.0.1:$port/okta-powershell/?test=response&headers=$headers&statuscode=429"
+
+        try {
+
+            { Start-HTTPListener -Port $port -Verbose} | Should -Not -Throw
+            
+            # Send a test request to verify the server is running
+            Start-Sleep -Seconds 2 # Allow server to start
+
+            $response = Invoke-WebRequest -Uri $url -SkipHttpErrorCheck
+            $response.StatusCode | Should -Be 429
+            $response.Headers['X-Okta-Request-Id'] | Should -Be "foo"
+            $response.Headers['X-Rate-Limit-Reset'] | Should -Be "bar"
+        }
+        finally {
+            Stop-HTTPListener -Port $port
+        }
+    }
+}
+
+Context 'Invoke-OktaApiClient should use HttpListener when calling IWR' {
+    BeforeAll {
+        
+        Import-Module -Name "$PSScriptRoot\..\..\..\src\Okta.PowerShell\Okta.PowerShell.PrivateFunctions.psd1" -Verbose
+        Import-Module -Name "$PSScriptRoot\HttpListener.psm1" -Verbose 
+    }
+
+    AfterAll {
+        Remove-Module -Name HttpListener -Verbose
+        Remove-Module -Name Okta.PowerShell.PrivateFunctions -Verbose
+    }
+
+    It 'OktaApiClient should use the HttpListener when calling IWR' {

```

Comment:
Is IWR a known concept?  What does it stand for?

Reply:
It refers to `Invoke-WebRequest`